src/modules/imageresizer/ui/Properties/Settings.cs

```diff
@@ -184,5 +209,221 @@ public void Reset()
+        public int JpegQualityLevel
+        {
+            get => _jpegQualityLevel;
+            set
+            {
+                _jpegQualityLevel = value;
+                NotifyPropertyChanged();
+            }
+        }
+
+        [JsonProperty(PropertyName = "imageresizer_pngInterlaceOption")]
+        public PngInterlaceOption PngInterlaceOption
+        {
+            get => _pngInterlaceOption;
+            set
+            {
+                _pngInterlaceOption = value;
+                NotifyPropertyChanged();
+            }
+        }
+
+        [JsonProperty(PropertyName = "imageresizer_tiffCompressOption")]
+        public TiffCompressOption TiffCompressOption
+        {
+            get => _tiffCompressOption;
+            set
+            {
+                _tiffCompressOption = value;
+                NotifyPropertyChanged();
+            }
+        }
+
+        [JsonProperty(PropertyName = "imageresizer_fileName")]
+        public string FileName
+        {
+            get => _fileName;
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    throw new System.ArgumentNullException();
+                }
+
+                _fileName = value;
+                NotifyPropertyChanged();
+            }
+        }
+
+        [JsonProperty(PropertyName = "imageresizer_sizes")]
+        public ObservableCollection<ResizeSize> Sizes
+        {
+            get => _sizes;
+            set
+            {
+                _sizes = value;
+                NotifyPropertyChanged();
+            }
+        }
+
+        [JsonProperty(PropertyName = "imageresizer_keepDateModified")]
+        public bool KeepDateModified
+        {
+            get => _keepDateModified;
+            set
+            {
+                _keepDateModified = value;
+                NotifyPropertyChanged();
+            }
+        }
+
+        [JsonProperty(PropertyName = "imageresizer_fallbackEncoder")]
+        public System.Guid FallbackEncoder
+        {
+            get => _fallbackEncoder;
+            set
+            {
+                _fallbackEncoder = value;
+                NotifyPropertyChanged();
+            }
+        }
+
+        [JsonProperty(PropertyName = "imageresizer_customSize")]
+        public CustomSize CustomSize
+        {
+            get => _customSize;
+            set
+            {
+                _customSize = value;
+                NotifyPropertyChanged();
+            }
+        }
+
+        public static string SettingsPath { get => _settingsPath; set => _settingsPath = value; }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        private void NotifyPropertyChanged([System.Runtime.CompilerServices.CallerMemberName] string propertyName = "")
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+        public void Save()
```

Comment:
IMHO, making changes to this can introduce minor JSON formatting errors if not handled properly. JSON data can be a C# Object converted to JSON object usong Newton.JSON.

Reply:
I had to implement my own Save/Reload functions as the original ApplicationSettings class which was being used only supports XML settings, but we want to migrate to JSON to match the other powertoys and keep it in the same format, i.e.
    {
        "version":1.0,
        "name": "ImageResizer",
        " properties": {
                                   imageresizer_someThing: { "value": 5.0 }, ...

If there is anything specific in the code that is concerning, do let me know

Reply:
Im sure the code wont break if you dont change this. Its just to make the code more cleaner and modularised.

Reply:
I've used the Newtonsoft.JSON library for converting the object in these lines:

    string tempJsonData = JsonConvert.SerializeObject(this);
    JObject tempSettings = JObject.Parse(tempJsonData);

The only string that is added to the JSON file without using the JSON library is "{\"version\":\"1.0\",\"name\":\"ImageResizer\",\"properties\":" which is to match the format that's used in the PT settings.json files. If I don't use the string I would have to either create a class structure to match the format or I would have to modify the JObject (which is what is being done for adding the value tags). I've added tests to ensure that the file created by Save() is readable by Reload(). Is there a clean simple alternate way to do this?